### PR TITLE
code clean up and renaming

### PR DESCRIPTION
**Pull request by panyx0718**
Reduce one level of inheritence.

### CODE REVIEW

File: paddle/fluid/framework/details/multi_devices_graph_pass.cc [hunk 1]
```diff
@@ -33,6 +33,92 @@
...
+  for (auto &op : graph->Get<GraphOps>(kGraphOps)) {
+    if (!op->Outputs().empty()) {
+      continue;
+    }
+    auto *dummy_leaf = new DummyVarHandle(graph->CreateControlDepVar());
+    graph->Get<GraphDepVars>(kGraphDepVars).emplace(dummy_leaf);
+    op->AddOutput(dummy_leaf);
+  }
+}
+}  // namespace
```
Comment (chengduoZH) [inline paddle/fluid/framework/details/multi_devices_graph_pass.cc:121]: I am curious about why you don't move the functions of the anonymous namespace to `MultiDevSSAGraphBuilder `?

File: paddle/fluid/framework/details/multi_devices_graph_pass.cc [hunk 2]
```diff
@@ -33,6 +33,92 @@
...
+        continue;
+      }
+      auto it_new = name_pair.second.rbegin();
+      auto it_old = name_pair.second.rbegin();
+      ++it_old;
+      for (; it_old != name_pair.second.rend(); it_new = it_old, ++it_old) {
+        OpHandleBase *write_op = (*it_new)->GeneratedOp();
+        const auto &read_ops = (*it_old)->PendingOps();
+
+        for (auto *read_op : read_ops) {
```
Comment (chengduoZH) [inline paddle/fluid/framework/details/multi_devices_graph_pass.cc:50]: Should we handle WFW(write after write) here?

File: paddle/fluid/framework/details/multi_devices_graph_pass.cc (hunk 1 above)
Comment (panyx0718) [reply paddle/fluid/framework/details/multi_devices_graph_pass.cc:121]: It's cleaner.

If a function is not *really* a class member, it shouldn't be.

File: paddle/fluid/framework/details/multi_devices_graph_pass.cc (hunk 2 above)
Comment (panyx0718) [reply paddle/fluid/framework/details/multi_devices_graph_pass.cc:50]: Yes, but it's not the purpose of this PR.

File: paddle/fluid/framework/details/multi_devices_graph_pass.cc [hunk 3]
```diff
@@ -33,6 +33,92 @@
 namespace paddle {
 namespace framework {
 namespace details {
+namespace {
+void PolishGraphToSupportDataHazards(ir::Graph *graph) {
```
Comment (chengduoZH) [inline paddle/fluid/framework/details/multi_devices_graph_pass.cc:37]: I think `PolishGraphToSupportDataHazards` will be used in the `pass` of removing or adding nodes.

**Review by chengduoZH — APPROVED** [review]

**Pull request merged**